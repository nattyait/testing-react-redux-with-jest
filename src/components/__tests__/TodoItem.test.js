import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import TodoItem from '../TodoItem'

// Describe what to test
describe('TodoItem', () => {
  // Test scenario
  it('runs jest', () => {
    expect(5).toBe(5)
  })
  // Test scenario
  it('checks array', () => {
    expect([1, 2, 3]).toEqual([1, 2, 3])
  })
  describe('TodoItem', () => {
    it('matches its snapshot - not complete', () => {
      const component = shallow(<TodoItem text="Not complete" />)
      expect(toJson(component)).toMatchSnapshot()
    })
    it('render correct structure', () => {
      const component = shallow(<TodoItem text="Hello" />)
      expect(component.is('li')).toBe(true)
    })
  })
  it('has complete class if receving complete prop', () => {
    const component = shallow(
      <TodoItem text="Hello" complete />
    )
    expect(component.hasClass('complete')).toBe(true)
  })
})
