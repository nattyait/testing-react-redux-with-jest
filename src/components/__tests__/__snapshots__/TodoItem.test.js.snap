// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem TodoItem matches its snapshot - not complete 1`] = `
<li
  className=""
>
  Not complete
</li>
`;
